validator/src/commands/run/args.rs

```diff
@@ -1179,6 +1179,18 @@ pub fn add_args<'a>(app: App<'a, 'a>, default_args: &'a DefaultArgs) -> App<'a,
             .takes_value(true)
             .help("Number of bins to divide the accounts index into"),
     )
+    .arg(
+        Arg::with_name("estimated_accounts")
+            .long("estimated-accounts")
+            .value_name("NUMBER")
+            .validator(is_parsable::<usize>)
+            .takes_value(true)
+            .help(
+                "Estimated total number of accounts in the ledger. Used to pre-allocate HashMap \
+                 capacity in the accounts index (divided by number of bins). Helps reduce memory \
+                 allocations and overallocations during initialization and runtime.",
+            ),
```

Comment:
I'm not sure how I feel about this yet...

On one hand, perf improvements are great. On the other, supporting and communicating a good value here may be challenging. Operator education probably shouldn't limit us for choosing perf improvements though.

As a middle ground, wdyt about marking the cli arg as hidden and see how it goes? We can unhide it later. Going the other way is harder though.

---

Other than ☝️, can we rename the arg? I'd feel better if we scope this to the accounts index, and then also call it "initial accounts" or so.


```suggestion
        Arg::with_name("accounts_index_initial_accounts")
            .long("accounts-index-initial-accounts")
            .value_name("NUMBER")
            .validator(is_parsable::<usize>)
            .takes_value(true)
            .help("Pre-allocate the accounts index, assuming this many number of accounts")
            .hidden(..),
```

(and rename the vars as well: s/estimated/initial/g)